From 3035aef48f4c1be1d24289c2ba605bc9f56c63cd Mon Sep 17 00:00:00 2001
From: "Philip, Avinash" <avinashphilip@ti.com>
Date: Sat, 10 Mar 2012 16:59:55 +0530
Subject: [PATCH 158/172] ARM: OMAP2+: epwm: Time base clock tree node.

pwmssctrl register (in control module) that controls time base clock
needs to be configured for proper working of ePWM. Time base module is
part of ePWM IP, and it's clock enabled through control module.

Enabling time base clock is handled by adding clock tree node and by
flagging it with ENABLE_ON_INIT.

Signed-off-by: Philip, Avinash <avinashphilip@ti.com>
---
 arch/arm/mach-omap2/clock33xx_data.c |   32 ++++++++++++++++++++++++++++++++
 1 files changed, 32 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/clock33xx_data.c b/arch/arm/mach-omap2/clock33xx_data.c
index bb340f3..c88e4cf 100644
--- a/arch/arm/mach-omap2/clock33xx_data.c
+++ b/arch/arm/mach-omap2/clock33xx_data.c
@@ -1986,6 +1986,35 @@ static struct clk wdt1_fck = {
 	.recalc		= &omap2_clksel_recalc,
 };
 
+/*
+ * Provides clock definitions for enabling bits for Time base module in
+ * PWMSS ctrl register.
+ */
+
+static struct clk ehrpwm0_tbclk = {
+	.name		= "ehrpwm0_tbclk",
+	.enable_reg	= AM33XX_CONTROL_PWMSS_CTRL,
+	.enable_bit	= AM33XX_PWMSS0_TBCLKEN,
+	.ops		= &clkops_omap2_dflt,
+	.flags		= ENABLE_ON_INIT,
+};
+
+static struct clk ehrpwm1_tbclk = {
+	.name		= "ehrpwm1_tbclk",
+	.enable_reg	= AM33XX_CONTROL_PWMSS_CTRL,
+	.enable_bit	= AM33XX_PWMSS1_TBCLKEN,
+	.ops		= &clkops_omap2_dflt,
+	.flags		= ENABLE_ON_INIT,
+};
+
+static struct clk ehrpwm2_tbclk = {
+	.name		= "ehrpwm2_tbclk",
+	.enable_reg	= AM33XX_CONTROL_PWMSS_CTRL,
+	.enable_bit	= AM33XX_PWMSS2_TBCLKEN,
+	.ops		= &clkops_omap2_dflt,
+	.flags		= ENABLE_ON_INIT,
+};
+
 
 /*
  * clkdev
@@ -2165,6 +2194,9 @@ static struct omap_clk am33xx_clks[] = {
 	CLK(NULL,	"gpt6_ick",		&timer6_ick,		CK_AM33XX),
 	CLK(NULL,	"gpt7_ick",		&timer7_ick,		CK_AM33XX),
 	CLK(NULL,	"vtp_clk",		&vtp_clk,		CK_AM33XX),
+	CLK(NULL,	"ehrpwm0_tbclk",	&ehrpwm0_tbclk,	CK_AM33XX),
+	CLK(NULL,	"ehrpwm1_tbclk",	&ehrpwm1_tbclk,	CK_AM33XX),
+	CLK(NULL,	"ehrpwm2_tbclk",	&ehrpwm2_tbclk,	CK_AM33XX),
 };
 
 int __init am33xx_clk_init(void)
-- 
1.7.9.1

